src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 10;
+    private static final double INCREASE_ARRAY = 1.5;
+    private Object[] arraylist = new Object[ARRAY_SIZE];
+    private int size;
+
+    public ArrayList() {
+        this.arraylist = arraylist;
+        this.size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size + 1 <= arraylist.length) {
+            arraylist[size] = value;
+        } else if (size + 1 > ARRAY_SIZE) {
+            grow();
+            arraylist[size] = value;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAdd(index);
+        if (size + 1 == arraylist.length) {
+            grow();
+        }
+        System.arraycopy(arraylist,index,arraylist,index + 1,size - index);
+        arraylist[index] = value;
+        size++;
+    }
 
+    private void grow() {
+        Object [] newArray = new Object[ (int)(arraylist.length * INCREASE_ARRAY)];
+        System.arraycopy(arraylist,0,newArray,0,size);
+        arraylist = newArray;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int startIndexForList = size();
+        if (list != null && (size + list.size() <= arraylist.length)) {
+            for (int j = 0; j < list.size(); j++) {
+                add(list.get(j),startIndexForList + j);
+            }
+        } else if (list != null && size() + list.size() > arraylist.length) {
+            grow();
+            for (int j = 0; j < list.size(); j++) {
+                add(list.get(j), startIndexForList + j);
+            }
+        }
```

Comment:
call method add(T value) for each element in the list
it should be written in 3 lines

Reply:
Done.Could you pls make me clear: when i get such type of comment should have not send correction until i find solution with 3 lines? 
There are: 
condition - 1 line, grow() - 2 line, iteration in list cycle - 3 line, calling method add(T value) - 4 line.